src/network/address.rs

```diff
@@ -95,12 +87,26 @@ impl Decodable for Address {
     fn consensus_decode<D: io::Read>(mut d: D) -> Result<Self, encode::Error> {
         Ok(Address {
             services: Decodable::consensus_decode(&mut d)?,
-            address: addr_to_be(Decodable::consensus_decode(&mut d)?),
+            address: read_be_address(&mut d)?,
             port: u16::swap_bytes(Decodable::consensus_decode(d)?)
         })
     }
 }
 
+/// Read a big-endian address from reader.
+fn read_be_address<R: io::Read>(mut r: R) -> Result<[u16; 8], encode::Error> {
+    let mut address = [0u16; 8];
+    let mut buf = [0u8; 2];
+
+    for i in 0..8 {
+        if io::Read::read(&mut r, &mut buf)? != 2 {
+            return Err(encode::Error::ParseFailed("missing address bytes"));
+        }
+        address[i] = u16::from_be_bytes(buf);
```

Comment:
Why not `for word in &mut address` and then `*word = u16::from_be_bytes(buf);`? (feel free to come up with better var name)

Reply:
Ah nice, thanks. Implemented as suggested.